### PR TITLE
EWPP-2875: Add EC logo and remove the title from EC footers.

### PR DESCRIPTION
**Pull request by 22Alexandra**

### CODE REVIEW

File: oe_theme.theme [hunk 1]
```diff
@@ -1721,6 +1721,30 @@ function oe_theme_theme_suggestions_oe_corporate_blocks_eu_footer_alter(array &$
...
 function oe_theme_preprocess_oe_corporate_blocks_ec_footer(array &$variables) {
+  $site_logo_href = 'https://commission.europa.eu';
+  // Ensure the site logo falls back to english version if the language is
+  // not EU language.
+  $language_code = \Drupal::languageManager()->getCurrentLanguage()->getId();
+  if (!EuropeanUnionLanguages::hasLanguage($language_code)) {
+    if ($language_code === 'nb') {
+      // Set correct icon for Norwegian language.
+      $variables['current_language_id'] = 'no';
+    }
```
Comment (nagyad) [inline oe_theme.theme]: Do we have this somewhere else? This resembles as pt-pt to me. Can we make this logic reusable if that is the case?

File: oe_theme.theme [hunk 2]
```diff
@@ -1721,6 +1721,30 @@ function oe_theme_theme_suggestions_oe_corporate_blocks_eu_footer_alter(array &$
...
+    else {
+      // Check whether ECL provides a logo for the language.
+      $ecl_logo = DRUPAL_ROOT . '/' . drupal_get_path('theme', 'oe_theme') . '/dist/' . $variables['ecl_component_library'] . '/images/logo/';
```
Comment (nagyad) [inline oe_theme.theme]: drupal_get_path is deprecated, let's use the service.

File: oe_theme.theme [hunk 3]
```diff
@@ -1721,6 +1721,30 @@ function oe_theme_theme_suggestions_oe_corporate_blocks_eu_footer_alter(array &$
...
+      $ecl_logo .= 'logo-' . $variables['ecl_component_library'] . '--' . $language_code . '.svg';
+      if (!file_exists($ecl_logo)) {
+        // Provide fallback "en" icon if language logo doesn't exist.
+        $variables['current_language_id'] = 'en';
+      }
+    }
+  }
+  else {
+    $site_logo_href .= '/index_' . EuropeanUnionLanguages::getInternalLanguageCode($variables['current_language_id']);
+  }
```
Comment (nagyad) [inline oe_theme.theme]: I see that we are executing the same code over the places. Please check if we can make a local helper function or a trait to reuse this logic.

**Review by nagyad — CHANGES_REQUESTED** [review]

**Review by nagyad — DISMISSED** [review]

**Review by nagyad — DISMISSED** [review]

File: tests/src/Behat/OeThemeTestContext.php [hunk 4]
```diff
@@ -284,29 +284,13 @@ protected function getDemoPageByTitle(string $title): NodeInterface {
...
-  /**
-   * Assert given corporate footer presence on page.
-   *
-   * @Then I should see the European Union footer with link :link label :label image alt :img_alt title :img_title
-   */
-  public function assertEuropeanUnionFooterBlockOnPage(string $link, string $label, string $img_alt, string $img_title): void {
+  public function assertEuropeanFooterBlockOnPage(string $component_library, string $link, string $label, string $img_alt, string $img_title): void {
     // Make sure a corporate footer is present on the page.
     $this->assertSession()->elementExists('css', 'footer.ecl-site-footer');
     // European Union footer is presented.
```
Comment (sergepavle) [inline tests/src/Behat/OeThemeTestContext.php:292]: Currently, it is not anymore only the European Union footer.

**Review by sergepavle — DISMISSED** [review]

**Review by upchuk — APPROVED** [review]
Blind

**Review by yenyasinn — APPROVED** [review]

**Pull request merged**